src/test/scala/is/hail/utils/RowIntervalSuite.scala

```diff
@@ -65,6 +65,49 @@ class RowIntervalSuite extends TestNGSuite {
       Interval(Row(1, 2), Row(1, 3), true, true)))
   }
 
+  @Test def testLteqWithOverlap() {
+    val eord = pord.intervalEndpointOrdering
+    assert(eord.lteqWithOverlap(3)(
+      IntervalEndpoint(Row(0, 1, 5), 1), IntervalEndpoint(Row(0, 1, 5), -1)
+    ))
+    assert(eord.lteqWithOverlap(3)(
+      IntervalEndpoint(Row(0, 1), 1), IntervalEndpoint(Row(0, 1), 1)
+    ))
+    assert(!eord.lteqWithOverlap(3)(
+      IntervalEndpoint(Row(0, 1, 6), -1), IntervalEndpoint(Row(0, 1, 5), 1)
+    ))
+    assert(!eord.lteqWithOverlap(2)(
+      IntervalEndpoint(Row(0, 1, 5), 1), IntervalEndpoint(Row(0, 1, 5), -1)
+    ))
+    assert(eord.lteqWithOverlap(2)(
+      IntervalEndpoint(Row(0, 1, 5), -1), IntervalEndpoint(Row(0, 1, 5), -1)
+    ))
+    assert(!eord.lteqWithOverlap(1)(
+      IntervalEndpoint(Row(0, 1, 5), -1), IntervalEndpoint(Row(0, 1, 5), -1)
+    ))
+    assert(!eord.lteqWithOverlap(1)(
+      IntervalEndpoint(Row(0, 1, 2), -1), IntervalEndpoint(Row(0, 1, 5), -1)
+    ))
+    assert(eord.lteqWithOverlap(1)(
+      IntervalEndpoint(Row(0, 1), -1), IntervalEndpoint(Row(0, 1), -1)
+    ))
+    assert(eord.lteqWithOverlap(1)(
+      IntervalEndpoint(Row(0, 1, 5), -1), IntervalEndpoint(Row(0, 2, 3), -1)
+    ))
+    assert(!eord.lteqWithOverlap(0)(
+      IntervalEndpoint(Row(0, 1), -1), IntervalEndpoint(Row(0, 1), -1)
+    ))
+    assert(!eord.lteqWithOverlap(0)(
+      IntervalEndpoint(Row(0, 1, 5), -1), IntervalEndpoint(Row(0, 2, 3), -1)
+    ))
+    assert(eord.lteqWithOverlap(0)(
+      IntervalEndpoint(Row(0), -1), IntervalEndpoint(Row(0), -1)
+    ))
+    assert(eord.lteqWithOverlap(0)(
+      IntervalEndpoint(Row(0, 3), -1), IntervalEndpoint(Row(1, 2), -1)
```

Comment:
let's add some tests with rows of different length

Reply:
I added a few more test cases, and reordered them in a way I think is clearer. Can you think of any more cases I should add, from either a testing or documentation perspective?